### ISSUE TITLE
Frame problem with Medium

### ISSUE THREAD

**Issue by fractaledvisions** [LibreVR/Revive#359]
I'm getting a display problem when I try to use Medium. When I move my HMD around, the frames get very shaky. Not slow, but shaky, flashing slightly out of place in varying opacity. If I stand still, the problem is barely noticeable. It is important to note that the viewer on my desktop doesn't show any shakiness or frame issues at all. I'm using a NVidia 980ti that performs great with all my Vive software, and I was able to run all the default Oculus Home software without any problems.

**Comment by LibreVR** [comment]
Sounds like a problem with reprojection, can you try turning off reprojection?

**Comment by fractaledvisions** [comment]
I played with all that and some of the other SteamVR settings without any luck

**Comment by fractaledvisions** [comment]
Another user mentioned that turning off reprojection did solve the problem for them, but I am still experiencing the problem. Any other tips or ideas,

**Comment by fractaledvisions** [comment]
?

**Comment by LibreVR** [comment]
Please follow these instructions to report performance problems: https://github.com/LibreVR/Revive/issues/353#issuecomment-265434808

**Comment by fractaledvisions** [comment]
<img width="553" alt="frame timing oculus medium error" src="https://cloud.githubusercontent.com/assets/24448045/21032154/5f934f70-bd5c-11e6-86a4-a665708fa468.png">
[VRFrames.txt](https://github.com/LibreVR/Revive/files/640854/VRFrames.txt)

**Comment by fractaledvisions** [comment]
Did the log file upload correctly?

**Comment by fractaledvisions** [comment]
[VRFrames.txt](https://github.com/LibreVR/Revive/files/640858/VRFrames.txt)

**Comment by fractaledvisions** [comment]
I'm on an i7 6700 4.0 Ghz, if it helps

**Comment by LibreVR** [comment]
@fractaledvisions You did it correctly. That graph looks insanely bad, your GPU is mostly idle, but your CPU is completely blocked. I'll look over the data during the weekend but I think that the postponed wait for early start is failing here.

**Comment by fractaledvisions** [comment]
Thanks for your help. Let me know if there's anything else I can try to get it working.

**Comment by LibreVR** [comment]
Please try this version, let me know if it fixes the issue: https://github.com/LibreVR/Revive/releases/download/1.0.3/ReviveInstaller_gl-no-postpone.exe

**Comment by fractaledvisions** [comment]
After my initial try the problem still exists. Going to uninstall/reinstall Revive again but initial test is no good :(

**Comment by fractaledvisions** [comment]
Uninstalled/reinstalled, restarted, double checked that I had turned off the Reprojection settings in SteamVR, still experiencing the problem

**Comment by LibreVR** [comment]
Ok, well good to know it's not the postponed wait. Then it's probably this "other" work on the GPU that's pushing us into late start.

Are you using an AMD GPU? Did you update your drivers?

**Comment by fractaledvisions** [comment]
NVidia 980ti with up to date drivers

**Comment by LibreVR** [comment]
Did you customize any global 3D settings in the Nvidia Control Panel?

**Comment by fractaledvisions** [comment]
I don't believe so. Should I ? I will double check but I am pretty certain I haven't played with any Nvidia settings on this computer

**Comment by fractaledvisions** [comment]
Nothing looks out of ordinary in my NVidia control panel, I don't have any custom settings for Medium or Revive or anything...

**Comment by LibreVR** [comment]
Then I'm not really sure what the issue is. There seems to be a lot of extra work on the GPU that is not accounted for in either the application (this includes Revive) or the compositor.

**Comment by LibreVR** [comment]
According to the SteamVR documentation this can be caused by either a major delay when drawing the companion (mirror) window on the desktop or another GPU-heavy application running in the background.

**Comment by hyperseven** [comment]
have you checked for any bitcoin/miner malware running?

**Comment by LibreVR** [comment]
@hyperseven If that were the case you would expect he'd have problems in any VR application.

**Comment by fractaledvisions** [comment]
I have good antivirus running and this computer is pretty clean in terms of installed software...

**Comment by fractaledvisions** [comment]
Now I can't even launch Medium, it just get's stuck in the Revive Dashboard with a spinning circle trying to launch, SteamVR says "(unresponsive) OculusMedium.exe" . Have restarted twice and ensured I have no sketchy background tasks running. 

@LibreVR I am going to bed now, I will try to set up GPUView when I get up. I have GPU-Z installed, does that help at all?  Also I thought we determined the CPU was having trouble, not the Gpu? Thanks for your help so far, I really appreciate you trying to help me get this working.

**Comment by LibreVR** [comment]
@fractaledvisions I thought so initially, but it doesn't look like it. If you want to be sure, go to the SteamVR Settings -> Developer and check "Always on reprojection". If that doesn't fix it, then it's not a problem on the CPU side.

**Comment by fractaledvisions** [comment]
Okay, I've used GPU View to take a log file, I think. Here's a dropbox link to the Merged.etl file. Let me know if I've done something incorrectly. Thanks. https://www.dropbox.com/s/7vfni2wo428h4z5/Merged.etl?dl=0

**Comment by LibreVR** [comment]
I've downloaded the file and removed the link from your comment as it contains some private information like which applications you have running.

**Comment by fractaledvisions** [comment]
@LibreVR have you been able to look at this issue at all?

**Comment by LibreVR** [comment]
@fractaledvisions Sorry, not yet, GPUView logs are difficult to interpret so it will take a bit of time.

**Comment by fractaledvisions** [comment]
Of course, at your convenience. I appreciate you getting back to me.

**Comment by asampal** [comment]
I'm getting the same problem as @fractaledvisions with Medium. Running with a GTX 1080 and 6770K CPU and also see smooth rendering on the desktop display of the app. The display mirror, on the other hand, shows the stuttering.

I also see the same behavior in Oculus Quill, but only under certain conditions. In Quill, there are some sample scenes that can be loaded as well as some tutorial sketches. The stuttering doesn't happen in the tutorial sketches, but it's there in the other sample scenes. I was trying to reproduce the stuttering while starting from a new drawing and managed to do it even with only a few strokes drawn. What I noticed is that if I used a really large translucent brush and drew a few strokes, there would be stuttering. So the problem seems to occur when trying to render large, translucent polygons. When I deleted these strokes, the stuttering stopped.

@LibreVR , maybe the above info can help you figure out where to look.

**Comment by asampal** [comment]
I just tried Medium and Quill with asynchronous reprojection turned off and interleaved reprojection on. Medium was fine like this - no stuttering. Quill, on the sample scenes that stuttered with async reprojection on, stuttered much less, though in areas of the scene where there was higher complexity, there was still some stuttering. I tried turning off interleaved reprojection as well as async, but the stuttering in Quill was now worse than with just async off.

**Comment by lobsterhands** [comment]
I have a 4790K and GTX 980 and Medium gives me the same tracking stuttering despite any changes to reprojection.

**Comment by chrisbraeuer411** [comment]
I have the same Problems with Medium. (i7, Gtx 1080, 16 ram) It seems it gets worse the more you skulpt there. The stuttering gets also stronger the closer you get to the skulpt. Try loading the robot shop sample. Stutters as hell. I would love to use Medium but as it is now its pain if you really want to use it. ( not just playing around) Please tell me if i can do something on my side to help track the problem. I dont want to buy a rift and touch just for 1 app.

**Comment by Armada651** [comment]
I'm still looking into this, please bear with me as there are some other problems I also need to deal with. I always deal with the low-hanging fruit first, but I expect I'll be able to look deeper into the problem this month.

**Comment by chemb0t** [comment]
I can also confirm observations from @asampal. Turning off async and interleaved reprojection on my machine (i7, GTX 1080, 64gb) reduces stuttering/jitter but does not eliminate it. A quick way to test is opening the main save/load menu and trying to scroll (use grip button) on either of the side panels. 

I am seeing similar behavior in Quill as well. The sample scenes stutter, especially the Museum scene where there are a lot of translucent strokes. Please let me know if there is more info I can provide.

**Comment by papatosaur** [comment]
Same lags for me on Gtx980Ti, any news ?

**Comment by zfarm** [comment]
I'm also getting the shutter using a gtx 1070, I'll have to check on cpu if that's important at this point. This may be irrelevant but also starring the sculpt always with mirror on...Will mess with rez and mirror settings

**Comment by dlazares** [comment]
Had the same problem today. Disabled reprojection and worked for me with an i7 and gtx 970

**Comment by asampal** [comment]
@dlazares , IIRC turning off async reprojection stops judder at low model complexity, but it's still there when it goes up.

**Comment by jmbihorel** [comment]
I have the same problem, when I stay in a small part of the main box, (maybe a quarter) I can sculpt without any problems, but as soon as I start to get higher resolutions, it starts to make a lot of drop frames.

**Comment by asampal** [comment]
@Armada651 , any progress on this issue?

**Comment by goddard11** [comment]
I'm looking for solutions to this as well, major frame rate drops when sculpting higher resolutions and when getting close to even low resolution geometry.  I'm on a 4790K with a GTX 1080.  I'd love to use medium more heavily with the vive,  I don't want to buy an oculus on top of my vive!  There's nothing as good as medium for sculpting right now for vive.
